test/gzip.js

```diff
@@ -56,6 +59,11 @@ test('handles gzip error', async t => {
 	t.is(err.name, 'ReadError');
 });
 
+test('decompress option opts out of decompressing', async t => {
+	const response = await got(s.url, {decompress: false, encoding: null});
```

Comment:
This doesn't match the docs. Here you explicitly set `encoding: null`, but in the docs you say it's returned as a compressed Buffer when `decompress: false`.